mdsuite/file_io/lammps_trajectory_files.py

```diff
@@ -57,222 +62,247 @@
 }
 
 
-class LAMMPSTrajectoryFile(TrajectoryFile):
+class LAMMPSTrajectoryFile(mdsuite.file_io.file_read.FileProcessor):
     """
-    Child class for the lammps file reader.
-
-    Attributes
-    ----------
-    obj : object
-            Experiment class instance to add to
-    header_lines : int
-            Number of header lines in the file format (lammps = 9)
-    file_path : str
-            Path to the trajectory file.
+    Reader for LAMMPS files
     """
 
-    def __init__(self, obj, header_lines=9, file_path=None, sort: bool = False):
-        """
-        Python class constructor
+    def __init__(
+        self,
+        file_path: str,
+        trajectory_is_sorted_by_ids=True,
+        custom_data_map: dict = None,
+    ):
         """
 
-        super().__init__(
-            obj, header_lines, file_path, sort=sort
-        )  # fill the experiment class
-
-        self.f_object = open(self.file_path)  # file object
-
-    def _get_number_of_atoms(self):
+        Parameters
+        ----------
+        file_path:
+            path to the LAMMPS trajectory file
+        trajectory_is_sorted_by_ids:
+            Flag to indicate if the particle order in the trajectory file is the same for each configuration
+        custom_data_map
+            If your file contains columns with data that is not part of the standard set of properties (see var_names),
+            you can map the column names to the corresponding property.
+            example: custom_data_map = {"Reduced_Momentum": ["rp_x", "rp_y", "rp_z"]}, if the file contains columns
+            labelled as 'rp_{x,y,z}' for the three components of the reduced momentum vector
         """
-        Get the number of atoms
-
-        Returns
-        -------
-
-        # user custom names for variables.
-        if rename_cols is not None:
-            var_names.update(rename_cols)
-
+        self.file_path = pathlib.Path(file_path).resolve()
+        self.n_header_lines = 9
+        self.trajectory_is_sorted_by_ids = trajectory_is_sorted_by_ids
+        if custom_data_map is None:
+            custom_data_map = {}
+        self.custom_data_map = custom_data_map
+
+        # attributes that will be filled in by get_metadata() and are later used by get_configurations_generator()
+        self._batch_size = None
+        self._id_column_idx = None
+        self._n_particles = None
+        self._species_dict = None
+        self._property_dict = None
+        self._mdata = None
```

Comment:
they seem to be similar across FileProcessors. Would it make sense to put them in the parent class?

Reply:
with regards to all the parent class arguments: I agree that a "TabularTextFileProcessor" could unite a few of the methods common to extxyz and lammpstraj. I will sneak in an intermediate class (or do it with common free functions, I will see which is best)